pylutron_caseta/smartbridge.py

```diff
@@ -294,10 +294,50 @@ async def set_value(
         """
         device = self.devices[device_id]
 
+        # Handle keypad LEDs which don't have a zone ID associated
+        if device.get("type") == "KeypadLED":
+            target_state = "On" if value > 0 else "Off"
+            await self._request(
+                "UpdateRequest",
+                f"/led/{device_id}/status",
+                {"LEDStatus": {"State": target_state}},
+            )
+            return
+
+        # All other device types must have an associated zone ID
         zone_id = device.get("zone")
         if not zone_id:
             return
 
+        # Handle Ketra lamps
+        if device.get("type") == "SpectrumTune":
+            if fade_time is not None:
+                await self._request(
+                    "CreateRequest",
+                    f"/zone/{zone_id}/commandprocessor",
+                    {
+                        "Command": {
+                            "CommandType": "GoToSpectrumTuningLevel",
+                            "SpectrumTuningLevelParameters": {
+                                "Level": value,
+                                "FadeTime": _format_duration(fade_time),
```

Comment:
In this case, since it's the same command for both, it'd probably make sense to do something like `if fade_time is not None: params["FadeTime"] = _format_duration(fade_time)` and just have one code block for invoking the command.

Reply:
Good suggestion – updated